File: day12/Day12.java
class Day12 {
    public int lastStoneWeight(int[] stones) {
        
        int n = stones.length;
        PriorityQueue<Integer>que = new PriorityQueue<Integer>();
        for(int i = 0 ;i < n; i++) {
            que.add(-stones[i]);
        }
        
        while(que.size() > 1) {
            
            int stone1 = que.poll();
            int stone2 = que.poll();
            int stone = Math.abs(stone1 - stone2);
            //System.out.println(stone1 + " "+stone2+" "+stone);
            if(stone > 0) {
                que.add(-stone);
            }
        }
        int ans = 0;
        if(que.size() > 0) {
            ans = que.poll();
        }
        return -ans;
    }
}

File: day12/day12.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        
        int ans = 0;
        priority_queue<int>que;
        for(int i = 0;i < stones.size();i++) {
            que.push(stones[i]);
        }
        while(que.size() > 1) {
            
            int num1 = que.top();
            que.pop();
            int num2 = que.top();
            que.pop();
            int num = abs(num1 - num2);
            if(num != 0) {
                que.push(num);
            }
        }
        if(que.size() == 1) {
            ans =  que.top();
        }
        return ans;
    }
};